Neuro/Tasks/TasksHandler.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections;
+using Neuro.Utilities;
+using Reactor.Utilities.Attributes;
+using UnityEngine;
+
+namespace Neuro.Tasks;
+
+[RegisterInIl2Cpp]
+public class TasksHandler : MonoBehaviour
+{
+    public TasksHandler(IntPtr ptr) : base(ptr) { }
+
+    // TODO: Move this to movement class
+    public Vector2[] CurrentPath { get; set; } = Array.Empty<Vector2>();
+    public int PathIndex { get; set; } = -1;
+
+    public IEnumerator EvaluatePath(NormalPlayerTask initial)
+    {
+        CurrentPath = NeuroPlugin.Instance.PathfindingHandler.FindPath(PlayerControl.LocalPlayer.transform.position, initial.Locations.At(0));
+        PathIndex = 0;
+
+        while (true)
+        {
+            yield return new WaitForSeconds(1);
+
+            PlayerTask task = PlayerControl.LocalPlayer.myTasks.At(0);
+
+            // TODO: Get the nearest location from all tasks instead of the first location of the next task
+            PlayerTask nextTask = null;
+            if (task.IsComplete)
+            {
+                Info("Task is complete");
+                foreach (PlayerTask t in PlayerControl.LocalPlayer.myTasks)
+                {
+                    if (!t.IsComplete && t.HasLocation)
+                    {
+                        nextTask = t;
+                        Info(nextTask.name);
+                        break;
+                    }
+                }
+            }
+            else
+            {
+                nextTask = task;
+            }
+
+            if (nextTask != null)
+            {
+                Info("Next task isn't null");
+                CurrentPath = NeuroPlugin.Instance.PathfindingHandler.FindPath(PlayerControl.LocalPlayer.transform.position, nextTask.Locations.At(0));
+                PathIndex = 0;
+
+                //pathfinding.DrawPath(currentPath);
+            }
+        }
+    }
+
+    private void Update()
```

Comment:
This Update should not be executed while game is not started. Same for RecordingHandler's and VisionHandler's Updates
`if (ShipStatus.Instance == null) return;`

Reply:
This is the best way to check if a game is ongoing. There is also a field in AmongUsClient for the state of the game but this is better. However we should use `if (!ShipStatus.Instance) return;` because an object that has been marked for destruction using GameObject.Destroy but has not yet been destroyed by unity will not be null, but it will still be "falsy".